docs/documentation/quartz-3.x/tutorial/using-quartz.md

```diff
@@ -12,70 +12,83 @@ in the paused state.
 
 Here's a quick snippet of code, that instantiates and starts a scheduler, and schedules a job for execution:
 
-__Using Quartz.NET__
+### Install Quartz.NET NuGets
+
+```sh
+Install-Package Microsoft.Extensions.Hosting
```

Comment:
I had to add this one oddly to my project to get it to find the `Host` class. 🤔

Reply:
yeah, such direct reference might need one. in a way it's of course good to list everything needed, but also might make it look complex. compared to some other solutions Quartz has tried to keep dependencies to minimum (see main Quartz package which also supports DB persistence), but nowadays probably would be OK to have a lot more dependencies especially if they are from "Microsoft dependency chain".

Reply:
I'll try it again and see what I can get. I can prolly remove the JSON reference too. Right?

Have you considered going all in on the `Microsoft.Extensions.Hosting` for v4? I'm of the opinion that all new projects are working this way.

That could let you get rid of `Quartz.Extensions.DependencyInjection`, `Quartz.Extensions.Hosting`

```csharp
var builder = Host.CreateDefaultBuilder()
    // makes quartz feel like a top level concern
    .UsingQuartz((cxt, options) =>
    {
        // does the "UseMicrosoftDependencyInjectionJobFactory" by default
        // services.AddQuartzHostedService();
    })
    .Build();
```

Reply:
> I'll try it again and see what I can get. I can prolly remove the JSON reference too. Right?

JSON probably needed, but here's the gist: I'd like people to know about the extension and use it (OK Quartz DI examples are full of using it, and so it should be). Another point is that it's sadly it's *the JSON package* instead of STJ and NSJ separation. 

This probably needs to be addressed in v4 timeframe or even 3.x. Deprecate/redirect JSON to `.Newtonsoft` and somehow endorse the STJ as default, even though the default name is already being used. I'm also quite scared of the migration path as people have data stored in database based on serialization format that also supports inheritance, e.g. `$type`-

> Have you considered going all in on the Microsoft.Extensions.Hosting for v4? I'm of the opinion that all new projects are working this way.

Yes, definitely. `Microsoft.Extensions.Hosting` (MEH?) is natural addition for such service like Quartz. I'd like to see Quartz 4 as meant to / designed for NET 6 and forward era, while still supporting full framework.

Reply:
Moving forward I think you can have two packages

`Quartz.Serialization.Newtonsoft`
`Quartz.Serialization.STJ`


That let's you leave the old "JSON" alone, and it doesn't survive into v4. For the v3 -> v4. You could have a 
 `Quartz.Serialization.UpgradeSupport` or some such that can be used to read the old format and rewrite it to the new format? or something. 🤔 

Also in a v3, version start detecting those issues and dropping warnings in the logs. ?? You could have some code that would attempt to upgrade to a newer model. Then in V4 you can do an upgrade. You could also start looking at a policy that you have to do v3 to v4, v4 to v5. No v3 to v5 type approach.

Reply:
Thanks for the ideas, definitely something to look into. `Quartz.Serialization.Json` could probably be an empty package that depends on `Quartz.Serialization.Newtonsoft` for deprecation. For greenfield projects I would probably already recommend using 3.x with (non-existent) `Quartz.Serialization.STJ` and gather feedback. What I've heard from serialization heavy projects like Marten, it might not be all that easy in all cases.